### DISCUSSION TITLE
HttpClientSession KeepAlive

### DISCUSSION THREAD

**Discussion by micheleselea** [pocoproject/poco#3794] [General]
I found a behavior I did not expect with HTTP keepalive connections. When (for example) a GET connection is triggered (with an HTTPS connection in that use case), if keepalive is true than a _reconnect is set to true so in function:
```
int HTTPClientSession::write(const char* buffer, std::streamsize length)
{
	try
	{
		int rc = HTTPSession::write(buffer, length);
		_reconnect = false;
		return rc;
	}
	catch (IOException&)
	{
		if (_reconnect)
		{
			close();
			reconnect();
			int rc = HTTPSession::write(buffer, length);
			clearException();
			_reconnect = false;
			return rc;
		}
		else throw;
	}
}
```
if something bad happen in write we try to reconnect. But I end up in a HTTPSession::write that returns -1 without trigger the exception so my send fail without a retry. Do you think was the design behavior?

**Comment by aleks-f** [comment]
I guess the rationale is that write returning -1 is not a connection problem, so trying to reconnect in such case is pointless

**Comment by micheleselea** [comment]
Understand, I don't know if -1 was for connection lost or other issue, because is not easy to reproduce that kind of problem, anyway the connection was an https
